### PR TITLE
xlsx support

### PR DESCRIPTION
**Pull request by PeterChaplin**
Requires:
```
sudo apt-get install libexcel-writer-xlsx-perl
```

### CODE REVIEW

File: tcl/excel.tcl [hunk 1]
```diff
@@ -10,13 +10,25 @@ proc qc::excel_file_create {args} {
     # column_meta: nested dict, eg. {1 {class "foo" width 20} 5 {width 50}}
```
Comment (daniel-qcode) [inline tcl/excel.tcl]: Can you update comment to ```Creates an xls or xslx file ...``` or something along those lines

File: tcl/excel.tcl [hunk 2]
```diff
@@ -10,13 +10,25 @@ proc qc::excel_file_create {args} {
...
+        "xls" {
+            set writer "Spreadsheet::WriteExcel"
+        }
+        "xlsx" {
+            set writer "Excel::Writer::XLSX"
+        }
+        default {
+            error "Unsupported excel type"
+        }
+    }
```
Comment (daniel-qcode) [inline tcl/excel.tcl]: We should maybe raise an issue to upgrade all projects to use the new xlsx format then we could make this the default.

**Comment by daniel-qcode** [comment]
Looks good just a few comments to address.

Then we will also need a similar pull request for qcode-tcl-6 (master)

**Comment by daniel-qcode** [comment]
Could you also update ```qc::mime_file_extension``` and ```qc::mme_type_guess``` for .xlsx  files ?

**Comment by PeterChaplin** [comment]
Updated mime types and comments

File: tcl/excel.tcl (hunk 2 above)
Comment (bvw) [inline tcl/excel.tcl]: Can we make xlsx the default `$type` as part of this pr ?
The xls format may become deprecated in time too.

Comment (PeterChaplin) [inline tcl/excel.tcl]: This proc is also used by `/new_crawley_bin_nos.xls` for tlc-erp.
If we're not worried about that, I'm happy to make this change.

Comment (bvw) [inline tcl/excel.tcl]: Good catch ! 
Could we change the tlc-erp code to specify `$type` and rollout the new qcode-5 package with that ?
Will then also need to go into qcode-6.

**Comment by PeterChaplin** [comment]
I've made xlsx the default type.

**Pull request merged**